pep-0692.rst

```diff
@@ -0,0 +1,344 @@
+
+::
+
+    def foo(**kwargs: **Movie) -> None:
+        kwargs["name"].capitalize()      # OK!
+        kwargs["year"] + 42              # OK!
+        kwargs["name"] + kwargs["year"]  # WRONG! Unsupported operand types for + ("str" and "int")
+
+The ``kwargs`` parameter inside the function should be treated exactly the same way as any other ``TypedDict``.
+
+Function calls with standard dictionaries
+-----------------------------------------
+
+Calling a function that has ``**kwargs`` typed using the ``**kwargs: **Movie`` sytnax with a standard dictionary must raise an error. For example:
+
+::
+
+    def foo(**kwargs: **Movie) -> None: ...
+
+    movie: dict[str, str] = {"name": "Life of Brian", "year": 1979}
+    foo(**movie)  # WRONG! Movie is of type dict[str, str]
+
+    typedMovie: Movie = {"name": "The Meaning of Life", "year": 1983}
+    foo(**typedMovie)  # OK!
+
+
+However, if the dictionary is completely untyped it is up to the type checker whether to raise an error or not:
+
+::
+
+    def foo(**kwargs: **Movie) -> None: ...
+
+    movie = {"name": "Life of Brian", "year": 1979} 
+
+    foo(**movie)  # Error/Success dependant on the type checker.
+
+
+Keyword collisions
+------------------
+
+A ``TypedDict`` that is used to type ``**kwargs`` could potentially contain keys that are already defined in the function's signature. If the duplicate name is a standard argument, an error should be raised. If the duplicate name is a positional only argument, no errors should be raised. For example:
+
+::
+
+    def foo(name, **kwargs: **Movie) -> None: ...     # WRONG! "name" will always bind to the first parameter.
+
+    def foo(name, /, **kwargs: **Movie) -> None: ...  # OK! "name" is a positional argument, so **kwargs can contain a "name" keyword
+
+
+Required and non-required keys
+------------------------------
+
+By default all keys in a ``TypedDict`` are required. This behaviour can be overriden by setting the dictionary's ``total`` parameter as ``False``:
+
+::
+
+    class Movie(TypedDict, total=False):
+        title: str
+        year: int 
+
+    dictionary: Movie = {"title": "Life of Brian"}         # OK! "year" is not required.
+    dictionary = {"title": "Life of Brian", "year": 1979}  # OK!
+
+Moreover, PEP 655 introduced new type qualifiers - ``typing.Required`` and ``typing.NotRequired`` - that enable specyfing whether a particular key is required or not:
+
+::
+
+    class Movie(TypedDict):
+        title: str
+        year: NotRequired[int]
+
+When using a ``TypedDict`` to type ``**kwargs`` all of the required and non-required keys should correspond to required and non-required function keyword parameters. Therefore, if a required key is not supported by the caller, then an error must be raised.
+
+Assignment
+----------
+A function typed using the ``**kwargs: **Movie`` construct can be assigned to another callable type only if they are compatible. This can happen in the following cases:
```

Comment:
"assigned to another callable type" -- I think this could be clearer. You're talking about valid assignments under static type checking, not runtime assignments. Perhaps "validly assigned" or "pass type checking" or similar.